packages/flame/lib/src/components/component.dart

```diff
@@ -93,10 +305,71 @@ abstract class Component {
   ///   myImage = await gameRef.load('my_image.png');
   /// }
   /// ```
+  @mustCallSuper
   Future<void>? onLoad() => null;
 
+  Future<void>? _onLoadCache;
+
+  /// Since [onLoad] only should run once throughout a components lifetime it is
+  /// cached so that it can be reused when the parent component/game/widget
+  /// changes.
+  @internal
+  Future<void>? get onLoadCache => _onLoadCache ?? (_onLoadCache = onLoad());
+
+  /// Called after the component has successfully run [prepare] and [onLoad] and
+  /// called before the component is added to its new parent.
```

Comment:
too many called words :P
```suggestion
  /// before the component is added to its new parent.
```